ctypesgen/__main__.py

```diff
@@ -28,114 +27,85 @@ def find_names_in_modules(modules):
     return names
 
 
-def option_callback_W(option, opt, value, parser):
-    # Options preceded by a "-Wl," are simply treated as though the "-Wl,"
-    # is not there? I don't understand the purpose of this code...
-    if len(value) < 4 or value[0:3] != "l,-":
-        raise optparse.BadOptionError("not in '-Wl,<opt>' form: %s%s" % (opt, value))
-    opt = value[2:]
-    if opt not in ["-L", "-R", "--rpath"]:
-        raise optparse.BadOptionError("-Wl option must be -L, -R" " or --rpath, not " + value[2:])
-    # Push the linker option onto the list for further parsing.
-    parser.rargs.insert(0, value)
```

Comment:
While I tried to keep backwards compatibility in general, I decided to removed the `-W` because it only seems to be some bizarre sort of alias/wrapper around the `"-L", "-R", "--rpath"` options, and I'm not sure how to re-implement that with argparse.

Reply:
[`-Wl` is used to specify options to be passed to the linker.](https://stackoverflow.com/q/1258998/319204).
Maybe the purpose of this logic was to enable passing compilations flags used for the C-code as-is to ctypesgen.

Reply:
Ah, yes. I was confused.
Apparently optparse recognises an option by prefix, so that `-Wl,-L some_value` is accepted for the `-W` parameter? But I don't think that's possible with argparse?
Also, I don't understand the `rargs` bit yet - where does the value land? Who takes it over?

Reply:
I'm very confused by this code. Did it work in that state?

Reply:
Am not sure either.

**Maybe** we can have a "unit-test" (automated/manual) 
- that specifies a few valid and unexpected `-Wl` params 
- verifies the resulting output params list
- and ensures the behavior with this PR continues to match the previous one.

IMO, whether it was "broken" can be eventually reviewed (and fixed) separately from this migration to argparse.

Reply:
I already tested the `-Wl` option on master. It is broken indeed. The implementation can't work.

I don't think there's any point wasting time on attempting to port broken code. It's unclear what the intent behind the option was, and judging by the code comment, other developers already thought so, too.

If this ever worked, someone should file a bug report sharing a command and the version of a compatible ctypesgen, then we could re-implement it cleanly.

In the meantime, I think it's fine to just remove this mess.

Reply:
FYI, I browsed the history and could trace the introduction of `-Wl` to this commit from 2007 by `james82`:
https://github.com/ctypesgen/ctypesgen/commit/21fe3a8818d480cc142a878769cb13ed31c94e38

Less than a year later, it has been marked with "I don't understand this code" by `TimMaxw`:
https://github.com/ctypesgen/ctypesgen/commit/ece4eaaff0a8af3877a1081febccc48062272c06#diff-a21b093b271328e34b3a159b2d5528365dd3b8c8fb6dc0c728e634163e69a7cb

From what I can tell, the code has been carried around as-is ever since and was never touched again.
It might never have been working.

Reply:
Note for future reference:

The list of directories obtained by parsing `-Wl, ...` 
is required to be used when searching for other libraries. 

- Searching for libraries implemented [here](https://github.com/ctypesgen/ctypesgen/blob/5396d03e76ff0aab2df557bdc9bddbdd5909d304/ctypesgen/processor/pipeline.py#L66).
- Called as part of the main sequence [here](https://github.com/ctypesgen/ctypesgen/blob/5396d03e76ff0aab2df557bdc9bddbdd5909d304/ctypesgen/processor/pipeline.py#L52) (via [this](https://github.com/ctypesgen/ctypesgen/blob/5396d03e76ff0aab2df557bdc9bddbdd5909d304/ctypesgen/processor/__init__.py#L12)).